src/core/execute.c

```diff
@@ -3375,6 +3370,12 @@ static int exec_child(
                 }
         }
 
+        /* Apply working directory here, because the working directory might be on NFS and only the user running
+         * this service might have the correct privilege to change to the working directory */
+        r = apply_working_directory(context, params, home, needs_mount_namespace, exit_status);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Changing to the requested working directory failed: %m");
```

Comment:
this really sounds like a game of whack-a-mole to me... So far our policy was to do security restrictions last, under the assumption that we can only lose privs, not gain them. The NFS messiness corrupts that logic though, suddenly doing security first enables us to do other stuff we weren't allowed to do before. Now, this really smells to me as if sooner or later different stuff people want to do would require different orderings, and that ultimately in a way that cannot be solved properly due to cyclic ordering reqs...

But anyway, let's see where this goes. The above is just  a bad feeling I have about this, maybe in reality it's not going to be that bad...

Reply:
For what it's worth, the requirement is not NFS-specific – it can apply equally well to any network filesystem, as local root _isn't_ automatically remote root in most of them. (CIFS? Accounts, kerberos tickets. AFS? Tokens. FUSE? allow_root/allow_other is off by default.)